core/src/main/clojure/xtdb/sql/plan.clj

```diff
@@ -1633,11 +1635,11 @@
     (let [ve (-> (.exprPrimary ctx) (.accept this))
           data-type (-> (.dataType ctx) (.accept (->CastArgsVisitor env)))]
       (handle-cast-expr ve data-type)))
-  
+
   (visitAggregateFunctionExpr [{:keys [!aggs] :as this} ctx]
     (if-not !aggs
       (add-err! env (->AggregatesDisallowed))
-      (-> (.aggregateFunction ctx) (.accept this))))
+      (-> (.aggregateFunction ctx) (.accept (assoc this :agg-expr true)))))
```

Comment:
nitpick: `?` for bool?

Reply:
try `(assoc this :!subqs !agg-subqs)` :crossed_fingers: